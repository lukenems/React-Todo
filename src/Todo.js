import React from 'react';

class Todo extends React.Component {
  render() {
    return (
      <li>
        {this.props.task}
        <button onClick={(evt => this.props.removeTodo(this.props.id, evt))}>x</button>
        <button onClick={(evt => this.props.showEditForm(this.props.id, evt))}>Edit</button>
      </li>
    )
  } 
}

export default Todo