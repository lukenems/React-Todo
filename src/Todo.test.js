import React from "react";
import { shallow } from "enzyme";
import Todo from './Todo';


describe('<Todo /> rendering', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Todo task="chores" />);
  });

  describe('Todo item', () => {
    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});