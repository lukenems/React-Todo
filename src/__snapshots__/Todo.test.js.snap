// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> rendering Todo item renders correctly 1`] = `ShallowWrapper {}`;
